src/parser/ast/field_term.rs

```diff
@@ -4,27 +4,54 @@ use crate::ir::ir_component::IRComponent;
 pub struct FieldTerm {
     name: Box<dyn ASTTerm>,
     type_name: Box<dyn ASTTerm>,
+    type_arguments: Vec<Box<dyn ASTTerm>>,
 }
 
 impl ASTTerm for FieldTerm {
     fn generate_ir(&self) -> Box<IRComponent> {
-        let fields = vec![
+        let mut fields = vec![
             ("name", self.name.generate_ir()),
             ("type", self.type_name.generate_ir()),
             ("hasDefault", Box::new(IRComponent::new_bool(false))),
         ];
 
+        if self.type_arguments.len() > 0 {
+            fields.push(("typeArguments", self.generate_type_arguments_ir()))
+        }
```

Comment:
Could we extract fields building to a separate method, for instance, ```build_fields``` to make generate_ir do only one thing?